using-mobile-extensions/places-extension/places-monitoring-extension/using-your-own-monitor.md

```diff
@@ -1,4 +1,142 @@
+
+```objective-c
+- (void) startMonitoringGeoFences: (NSArray*) newGeoFences {
+	// verify if the device supports monitoring geofences
+	// check for location permission
+
+    for (ACPPlacesPoi * currentRegion in newGeoFences) {
+        // make the circular region
+        CLLocationCoordinate2D center = CLLocationCoordinate2DMake(currentRegion.latitude, currentRegion.longitude);
+        CLCircularRegion* currentCLRegion = [[CLCircularRegion alloc] initWithCenter:center                                                                              												radius:currentRegion.radius                                                                     											   identifier:currentRegion.identifier];
+        currentCLRegion.notifyOnExit = YES;
+        currentCLRegion.notifyOnEntry = YES;
+
+        // start monitoring the new region
+        [_locationManager startMonitoringForRegion:currentCLRegion];
+
+    }
+}
+```
+
+
+
+{% endtab %}
+
+{% tab title="Android" %}
+
+-  The location updates obtained from the google play services or the android location services should be passed to the Places Extension. Use the `getNearbyPointsOfInterest` Places Extension API to get the list of n  `PlacesPoi` objects around the current location.
+
+  ```java
+  	LocationCallback callback = new LocationCallback() {
+  			@Override
+  			public void onLocationResult(LocationResult locationResult) {
+  				super.onLocationResult(locationResult);
+                  
+                  Places.getNearbyPointsOfInterest(currentLocation, 10, new 		   AdobeCallback<List<PlacesPOI>>() {
+  			@Override
+  			public void call(List<PlacesPOI> pois) {
+  				starMonitoringGeofence(pois);
+  			}
+  		});
+  
+  			}
+  		};
+  ```
+
+- Extract the data from the obtained `PlacesPOI` objects and start monitoring those points of intrests.
+
+  ```java
+  private void startMonitoringFences(final List<PlacesPOI> nearByPOIs) {
+  	// check for location permission
+      for (PlacesPOI poi : nearByPOIs) {
+  			final Geofence fence = new Geofence.Builder()
+  			.setRequestId(poi.getIdentifier())
+  			.setCircularRegion(poi.getLatitude(), poi.getLongitude(), poi.getRadius())
+  			.setExpirationDuration(Geofence.NEVER_EXPIRE)
+  			.setTransitionTypes(Geofence.GEOFENCE_TRANSITION_ENTER |
+  								Geofence.GEOFENCE_TRANSITION_EXIT)
+  			.build();
+  			geofences.add(fence);
+  		}
+  
+  		GeofencingRequest.Builder builder = new GeofencingRequest.Builder();
+  		builder.setInitialTrigger(GeofencingRequest.INITIAL_TRIGGER_ENTER);
+  		builder.addGeofences(geofences);
+  		builder.build();
+     		geofencingClient.addGeofences(builder.build(), geoFencePendingIntent)
+  }
+  ```
+
+{% endtab %}
+
+{% endtabs %}
+
+**Important** : Calling the ``getNearbyPointsOfInterest` Places Extension API, results in a network call of getting the location around the current location. Please call the API sparingly or only upon a significant location change of the user.
```

Comment:
Please reword this note so that it reads as follows:

"Calling the ``getNearbyPointsOfInterest` API results in a network call that gets the location around the current location. 

**Important** : You should call the API sparingly or only when there is significant location change of the user."

(The important part is to tell users to call this API sparingly, so I took the first sentence out of the note.) Also, please never use "please" in technical documentation :-).